internal/reflect/slice.go

```diff
@@ -90,14 +90,71 @@ func reflectSlice(ctx context.Context, typ attr.Type, val tftypes.Value, target
 		}
 
 		return slice, diags
-	// Tuple reflection from slices is currently not supported as usage is limited
+	// Tuple reflection into slices is currently limited to use-cases where all tuple element types are the same.
 	case attr.TypeWithElementTypes:
-		diags.Append(diag.WithPath(path, DiagIntoIncompatibleType{
-			Val:        val,
-			TargetType: target.Type(),
-			Err:        fmt.Errorf("cannot reflect %s using type information provided by %T, reflection support is currently not implemented for tuples", val.Type(), t),
-		}))
-		return target, diags
+		// we need to know the type the slice is wrapping
+		elemType := target.Type().Elem()
+
+		// we want an empty version of the slice
+		slice := reflect.MakeSlice(target.Type(), 0, len(values))
+
+		if len(t.ElementTypes()) <= 0 {
+			// If the tuple values are empty as well, we can just pass back an empty slice of the type we received.
+			if len(values) == 0 {
+				return slice, diags
+			}
+
+			diags.Append(diag.WithPath(path, DiagIntoIncompatibleType{
+				Val:        val,
+				TargetType: target.Type(),
+				Err:        fmt.Errorf("cannot reflect %s using type information provided by %T, tuple type contained no element types but received values", val.Type(), t),
+			}))
+			return target, diags
+		}
+
+		// Ensure that all tuple element types are the same by comparing each element type to the first
```

Comment:
Does this need an exception for successfully working with `[]attr.Value` that should be able to accept multiple element types?

Reply:
This sparked an offline conversation about "why exactly we need to utilize a Tuple instead of a List".

I have attempted to distill that conversation into a maintainer comment. @bendbennett + @bflad if you could take a look at 6dfc0320832b, let me know if that makes things more/less confusing 😆

Reply:
Thanks for adding the maintainer comment, the expanded context is very useful.